src/rockstor/smart_manager/data_collector.py

```diff
@@ -23,6 +46,265 @@
+                reader_process = Popen(read_command, bufsize=1, stdout=PIPE)
+                log_content = reader_process.communicate()[0]
+                log_contentsize = getsizeof(log_content)
+                log_content = log_content.splitlines(True)
+                
+                #Starting from content num of lines decide if serve it 1 line/time or in 200 lines chunks
+                reader_type = 'fast' if (len(log_content) <= 200) else 'slow'
+                chunk_size = logs_loader[reader_type]['lines']
+                reader_sleep = logs_loader[reader_type]['sleep']
+                log_content_chunks = [log_content[x:x+chunk_size] for x in xrange(0, len(log_content), chunk_size)]
+                total_rows = len(log_content)
+
+            else:#Log file missing or size 0, gently inform user
+            
+                #Log not exist or empty so we send fake values
+                #for rows, chunks, etc to uniform data on existing functions
+                #and avoid client side extra checks
+                log_content = 'Selected log file is empty or doesn\'t exist'
+                log_content = log_content.splitlines(True)
+                total_rows = 1
+                log_contentsize = getsizeof(log_content)
+                log_content_chunks = []
+                log_content_chunks.append(log_content)
+                reader_sleep = 0
+            
+            #Serve each chunk with emit and sleep before next one to avoid client side browser overload
+            current_rows = 0
+            
+            for data_chunks in log_content_chunks:
+                chunk_content = ''.join(data_chunks)
+                current_rows += len(data_chunks)
+                self.emit('logManager:logcontent', {
+                    'key': 'logManager:logcontent', 'data': {
+                    'current_rows' : current_rows,
+                    'total_rows' : total_rows,
+                    'chunk_content' : chunk_content,
+                    'content_size' : log_contentsize
+                    }
+                })
+                gevent.sleep(reader_sleep)
+        
+        def live_reader(log_path):
+
+            #Switch live reader state to True
+            self.livereading = True
+            
+            #Build reader command from readers dict
+            read_command = build_reader_command('tailf')
+
+            self.livereader_process = Popen(read_command, bufsize=1, stdout=PIPE)
+            while self.livereading:
+                live_out = self.livereader_process.stdout.readline()
+                self.emit('logManager:logcontent', {
+                    'key': 'logManager:logcontent', 'data': {
+                    'current_rows' : 1,
+                    'total_rows' : 1,
+                    'chunk_content' : live_out,
+                    'content_size' : 1
+                    }
+                })
+                gevent.sleep(0)
```

Comment:
Can you please explain the rationale behind putting gevent.sleep(0) in the while loop here?